liquidation-monitor/src/agent.js

```diff
@@ -0,0 +1,389 @@
+
+// Stores information about each account
+const initializeData = {};
+
+// Global functions
+function createAlert(
+  developerAbbreviation,
+  protocolName,
+  protocolAbbreviation,
+  type,
+  severity,
+  borrowerAddress,
+  liquidationAmount,
+  shortfallAmount,
+  healthFactor,
+) {
+  return Finding.fromObject({
+    name: `${protocolName} Liquidation Threshold Alert`,
+    description: `The address ${borrowerAddress} has dropped below the liquidation threshold. `
+      + `The account may be liquidated for: $${liquidationAmount} USD`,
+    alertId: `${developerAbbreviation}-${protocolAbbreviation}-LIQUIDATION-THRESHOLD`,
+    type: FindingType[type],
+    severity: FindingSeverity[severity],
+    protocol: protocolName,
+    metadata: {
+      borrowerAddress,
+      liquidationAmount,
+      shortfallAmount,
+      healthFactor,
+    },
+  });
+}
+
+async function verifyToken(data, tokenAddressImport) {
+  const { tokens } = data;
+  const tokenAddress = tokenAddressImport.toLowerCase();
+  if (tokens[tokenAddress] === undefined) {
+    tokens[tokenAddress] = {};
+    tokens[tokenAddress].contract = new ethers.Contract(
+      tokenAddress, data.cTokenABI, data.provider,
+    );
+    const cContract = tokens[tokenAddress].contract;
+    tokens[tokenAddress].symbol = await cContract.symbol();
+
+    // cETH does not have an underlying contract, so link it to wETH instead
+    if (tokenAddress === '0x4ddc2d193948926d02f9b1fe9e1daa0718270ed5') {
+      tokens[tokenAddress].underlying = '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2';
+    } else {
+      tokens[tokenAddress].underlying = await cContract.underlying();
+    }
+    const exchangeRate = new BigNumber((await cContract.exchangeRateStored()).toString());
+    tokens[tokenAddress].cTokenDecimals = new BigNumber(
+      (await cContract.decimals()).toString(),
+    );
+
+    // Look up decimals of the underlying tokens as well.
+    const decimalsABI = '["function decimals() view returns (uint)"]';
+    const underlyingTokenContract = new ethers.Contract(
+      tokens[tokenAddress].underlying, decimalsABI, data.provider,
+    );
+    tokens[tokenAddress].tokenDecimals = new BigNumber(
+      (await underlyingTokenContract.decimals()).toString(),
+    );
+
+    tokens[tokenAddress].tokenDecimalsMult = new BigNumber(10)
+      .pow(tokens[tokenAddress].tokenDecimals);
+    tokens[tokenAddress].cTokenDecimalsMult = new BigNumber(10)
+      .pow(tokens[tokenAddress].cTokenDecimals);
+
+    // Adjusting the multiplier for easier use later.
+    // “The current exchange rate as an unsigned integer, scaled by
+    //   1 * 10 ^ (18 - 8 + Underlying Token Decimals)” - https://compound.finance/docs/ctokens#exchange-rate
+    //   Simplified to 10^(10 + Underlying Token Decimals).
+    const exchangeDecimals = new BigNumber(10).pow(10);
+    tokens[tokenAddress].exchangeRateMult = exchangeRate
+      .dividedBy(exchangeDecimals).dividedBy(tokens[tokenAddress].tokenDecimalsMult);
```

Comment:
To make this consistent with the variable naming elsewhere:
```suggestion
    const exchangeDecimalsMult = new BigNumber(10).pow(10);
    tokens[tokenAddress].exchangeRateMult = exchangeRate
      .dividedBy(exchangeDecimalsMult).dividedBy(tokens[tokenAddress].tokenDecimalsMult);
```